code/go/0chain.net/blobbercore/handler/storage_handler.go

```diff
@@ -325,56 +323,83 @@ func (fsh *StorageHandler) AddCollaborator(ctx context.Context, r *http.Request)
 	return result, nil
 }
 
-func (fsh *StorageHandler) GetFileStats(ctx context.Context, r *http.Request) (interface{}, error) {
-	if r.Method == "GET" {
-		return nil, common.NewError("invalid_method", "Invalid method used. Use POST instead")
-	}
-	allocationTx := ctx.Value(constants.ALLOCATION_CONTEXT_KEY).(string)
+func (fsh *StorageHandler) GetFileStats(ctx context.Context, request *blobbergrpc.GetFileStatsRequest) (interface{}, error) {
+	//if r.Method == "GET" {
+	//	return nil, common.NewError("invalid_method", "Invalid method used. Use POST instead")
+	//}
+	//allocationTx := ctx.Value(constants.ALLOCATION_CONTEXT_KEY).(string)
+	allocationTx := request.Allocation
 	allocationObj, err := fsh.verifyAllocation(ctx, allocationTx, true)
 	if err != nil {
-		return nil, common.NewError("invalid_parameters", "Invalid allocation id passed."+err.Error())
+		return nil, errors.Wrap(err, "invalid allocation id in the request")
 	}
 	allocationID := allocationObj.ID
 
-	clientSign, _ := ctx.Value(constants.CLIENT_SIGNATURE_HEADER_KEY).(string)
+	clientSign := ctx.Value(constants.CLIENT_SIGNATURE_HEADER_KEY).(string)
+
+	// validate the signature from request
 	valid, err := verifySignatureFromRequest(allocationTx, clientSign, allocationObj.OwnerPublicKey)
-	if !valid || err != nil {
-		return nil, common.NewError("invalid_signature", "Invalid signature")
+	if err != nil || !valid {
+		return nil, errors.Wrap(err, "invalid signature passed in request")
 	}
 
 	clientID := ctx.Value(constants.CLIENT_CONTEXT_KEY).(string)
-	if len(clientID) == 0 || allocationObj.OwnerID != clientID {
-		return nil, common.NewError("invalid_operation", "Operation needs to be performed by the owner of the allocation")
+	if (clientID == "") || (allocationObj.OwnerID != clientID) {
+		return nil, errors.Wrap(errors.New("Invalid Operation"), "operation needs to be performed by allocation's owner")
```

Comment:
it is better to move "InvalidOperation" to a variable if we want to wrap it.

e.g

```
var ErrInvalidOperation = errors.New("Invalid Operation")
```

Reply:
This is the next step when complete error package will be deprecated, as it is only duplicated code